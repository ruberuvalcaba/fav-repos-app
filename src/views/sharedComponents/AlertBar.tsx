import React from 'react'
import { Box, Alert, AlertIcon, AlertTitle, AlertDescription, CloseButton } from '@chakra-ui/react'

interface AlertBarProps {
  message: string
  description?: string
  status?: 'error' | 'info' | 'warning' | 'success' | 'loading'
  onClose: () => void
}

const AlertBar = ({
  message,
  status = 'error',
  description,
  onClose,
}: AlertBarProps): JSX.Element => {
  return (
    <Alert
      status={status}
      style={{
        display: 'flex',
        justifyContent: 'space-between',
        position: 'absolute',
        zIndex: '1',
      }}
    >
      <Box display="flex" justifyContent="space-between">
        <AlertIcon />
        <Box>
          <AlertTitle fontSize="sm">{message}</AlertTitle>
          <AlertDescription fontSize="xs">{description}</AlertDescription>
        </Box>
      </Box>
      <CloseButton
        alignSelf="flex-start"
        position="relative"
        right={-1}
        top={-1}
        onClick={onClose}
      />
    </Alert>
  )
}

export default AlertBar
