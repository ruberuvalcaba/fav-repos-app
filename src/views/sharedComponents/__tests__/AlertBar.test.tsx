import React from 'react'
import renderer from 'react-test-renderer'
import AlertBar from '../AlertBar'

it('AlertBar component renders correctly', () => {
  const component = renderer.create(
    <AlertBar
      message="Message Test"
      description="Description Test"
      status="error"
      onClose={() => {}}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
