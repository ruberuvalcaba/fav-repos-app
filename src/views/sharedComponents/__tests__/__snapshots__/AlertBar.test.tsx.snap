// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBar component renders correctly 1`] = `null`;
